_docs/developer/testing/cypress.md

```diff
@@ -0,0 +1,187 @@
+---
+title: End-to-End testing with Cypress
+---
+
+[Cypress](https://www.cypress.io/) is an open source end-to-end testing framework. It runs tests against 
+a locally hosted instance of Submitty by automating a browser. Tests are written in JavaScript and are executed 
+in the Cypress test-runner.
+
+## Installation 
+
+Make sure you have `node.js` version 10 or higher installed locally. You can check using the command: `node -v` 
+
+```bash
+# from the site directory
+npm install --include=dev
+```
+
+If you are on Linux you may need the following dependencies if they are not already installed:
+```bash
+apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+```
```

Comment:
Let's link to the official docs here so that we're not maintaining this list.

```suggestion
If you are on Linux, view [this page](https://docs.cypress.io/guides/getting-started/installing-cypress.html#Linux) to see what dependencies you may need to install additionally for Cypress.
```